lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -104,10 +108,21 @@ def make_ps_init_code(init_ready_event_name)
         template.result(binding)
       end
 
-      def make_ps_code(powershell_code, output_ready_event_name, timeout_ms = 300 * 1000)
+      def make_ps_code(powershell_code, output_ready_event_name, timeout_ms = default_execute_timeout, working_dir = nil)
+        # Convert the working directory to a native pathname, if required
+        if (working_dir == nil)
+          set_ps_location = ''
+        else
+          if Puppet::Util::Platform.windows?
+            working_dir = working_dir.gsub(File::SEPARATOR, File::ALT_SEPARATOR)
+          end
+          set_ps_location = "Set-Location \"#{working_dir}\" | Out-Null\n\r" unless working_dir == nil
+        end
+
         <<-CODE
 $params = @{
   Code = @'
+#{set_ps_location}
```

Comment:
I'm not sure how else to separate the two pieces of code.  A single line feed should be enough for PS to know the end of the `Set-Location....` and beginning of any other proceeding text.

Not sure about the failure.  Should that be terminating? probably.  Chances of it happening? Low.  There is an existance check in the provider so whatever path is passed in must exist.  Although permissions could be an issue i.e. user has rights to read that the directory exists, but not traversal.